sdk/nodejs/apigateway/experimental/api.ts

```diff
@@ -292,46 +315,72 @@ interface ApigatewayIntegration {
+        "x-amazon-apigateway-gateway-responses": {
+            "MISSING_AUTHENTICATION_TOKEN": {
+                "statusCode": 404,
+                "responseTemplates": {
+                    "application/json": "{\"message\": \"404 Not found\" }",
+                },
+            },
+            "ACCESS_DENIED": {
+                "statusCode": 404,
+                "responseTemplates": {
+                    "application/json": "{\"message\": \"404 Not found\" }",
+                },
+            },
+        },
     };
 
     // Now add all the routes to it.
-    addRoutesToSwaggerSpec(name, swagger, routes, opts);
-    addStaticRoutesToSwaggerSpec(name, swagger, staticRoutes, opts);
-    addProxyRoutesToSwaggerSpec(name, swagger, proxyRoutes, opts);
+    for (const path in routes) {
+        if (routes.hasOwnProperty(path)) {
+            const route = routes[path];
+
+            switch (route.kind) {
+                case "EventHandler":
+                    addEventHandlerRouteToSwaggerSpec(name, swagger, path, route, opts);
+                    continue;
+                case "Static":
+                    addStaticRouteToSwaggerSpec(name, swagger, path, route, opts);
+                    continue;
+                case "Proxy":
+                    addProxyRouteToSwaggerSpec(name, swagger, path, route, opts);
+                    continue;
+                default:
+                    const exhaustiveMatch: never = route;
+                    throw new Error('Non-exhaustive match for route');
+            }
+        }
+    }
 
     return swagger;
 }
 
-function addRoutesToSwaggerSpec(
-    name: string, swagger: SwaggerSpec, routes: Route[], opts: pulumi.ResourceOptions) {
-
-    if (!routes || routes.length === 0) {
-        return;
+function addSwaggerOperation(swagger: SwaggerSpec, path: string, method: string, operation: SwaggerOperation) {
+    if (!swagger.paths[path]) {
+        swagger.paths[path] = {};
     }
 
-    for (const route of routes) {
-        const method = swaggerMethod(route.method);
-        const lambdaFunc = lambda.createFunctionFromEventHandler(
-            name + sha1hash(method + ":" + route.path), route.handler, opts);
-        if (!swagger.paths[route.path]) {
-            swagger.paths[route.path] = {};
-        }
-        swagger.paths[route.path][method] = createSwaggerOperationForLambda(lambdaFunc);
-    }
+    swagger.paths[path][method] = operation;
+}
 
+function addEventHandlerRouteToSwaggerSpec(
+    name: string, swagger: SwaggerSpec, path: string, route: EventHandlerRoute, opts: pulumi.ResourceOptions) {
+
+    const method = swaggerMethod(route.method);
+    const lambdaFunc = lambda.createFunctionFromEventHandler(
+        name + sha1hash(method + ":" + path), route.eventHandler, opts);
+
+    addSwaggerOperation(swagger, path, method, createSwaggerOperationForLambda(lambdaFunc));
     return;
```

Comment:
same logic as before, just withotu the foreach loop over the routes since that is handled by our caller.